### PR TITLE
stage 1 java io theory init

### PR DESCRIPTION
**Pull request by Tiger656**

### CODE REVIEW

File: stage #1/module #6. IO, NIO/lesson #1. IO/content.md [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+- [Character Streams](https://docs.oracle.com/javase/tutorial/essential/io/charstreams.html) handle I/O of character data, automatically handling translation to and from the local character set.
+- [Buffered Streams](https://docs.oracle.com/javase/tutorial/essential/io/buffers.html) optimize input and output by reducing the number of calls to the native API.
+- [Scanning and Formatting](https://docs.oracle.com/javase/tutorial/essential/io/scanfor.html) allows a program to read and write formatted text.
+- [I/O from the Command Line](https://docs.oracle.com/javase/tutorial/essential/io/cl.html) describes the Standard Streams and the Console object.
+- [Data Streams](https://docs.oracle.com/javase/tutorial/essential/io/datastreams.html) handle binary I/O of primitive data type and String values.
+- [Object Streams](https://docs.oracle.com/javase/tutorial/essential/io/objectstreams.html) handle binary I/O of objects.
+
+No matter how they work internally, all streams present the same simple model to programs that use them: A stream is a sequence of data. A program uses an *input stream* to read data from a source, one item at a time:
+![img_1.png](assets/img_1.png)
```
Comment (egoist301) [inline stage #1/module #6. IO, NIO/lesson #1. IO/content.md:29]: Could you use names that will reflect the information of the file?

File: stage #1/module #6. IO, NIO/lesson #1. IO/content.md [hunk 2]
```diff
@@ -0,0 +1,127 @@
+# Java I/O (java.io)
+
+## Introduction to IO
+
+Java I/O (Input and Output) is used to process the input and produce the output.
+Java uses the concept of a stream to make I/O operation fast. The java.io package contains all the classes
```
Comment (egoist301) [inline stage #1/module #6. IO, NIO/lesson #1. IO/content.md:6]: Could you wrap package with `? For example `java.io`.

File: stage #1/module #6. IO, NIO/lesson #1. IO/content.md [hunk 3]
```diff
@@ -0,0 +1,127 @@
+# Java I/O (java.io)
```
Comment (egoist301) [inline stage #1/module #6. IO, NIO/lesson #1. IO/content.md:1]: Sorry, but module name is **IO, NIO**. Will you be doing another pull request for NIO?

File: stage #1/module #6. IO, NIO/lesson #1. IO/content.md [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+![img.png](assets/img.png)
+
+## Byte Streams
+
+Programs use *byte streams* to perform input and output of 8-bit bytes. All byte stream classes are descended from [InputStream](https://docs.oracle.com/javase/8/docs/api/java/io/InputStream.html) and [OutputStream](https://docs.oracle.com/javase/8/docs/api/java/io/OutputStream.html).
+There are many byte stream classes. To demonstrate how byte streams work, we'll focus on the file I/O byte streams, [FileInputStream](https://docs.oracle.com/javase/8/docs/api/java/io/FileInputStream.html) and [FileOutputStream](https://docs.oracle.com/javase/8/docs/api/java/io/FileOutputStream.html) which read and write files. Other kinds of byte streams are used in much the same way; they differ mainly in the way they are constructed.
+
+Following is an example which makes use of these two classes to copy an input file into an output file using byte stream, one byte at a time:
+
+| ![img_2.png](assets/img_2.png) |
```
Comment (egoist301) [inline stage #1/module #6. IO, NIO/lesson #1. IO/content.md:41]: Fix borders

File: stage #1/module #6. IO, NIO/lesson #1. IO/content.md [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+|-------------------------|
+| Byte Stream usage       |
+
+*CopyFile* spends most of its time in a simple loop that reads the input stream and writes the output stream, one byte at a time, as shown in the following figure.
+![img_4.png](assets/img_4.png)
+
+Method “read()” reads one element and return it’s value. When operation read is ended, then returned “-1”. Method “write(c)” writes one element into file.
```
Comment (egoist301) [inline stage #1/module #6. IO, NIO/lesson #1. IO/content.md:48]: Could you wrap part of code with `? (`read()`)

File: stage #1/module #6. IO, NIO/lesson #1. IO/content.md [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+
+Character streams are often "wrappers" for byte streams. 
+The character stream uses the byte stream to perform the physical I/O, 
+while the character stream handles translation between characters and bytes. *FileReader*, for example,
+uses *FileInputStream*, while *FileWriter* uses *FileOutputStream*. The major difference is that *FileReader* reads two bytes at one time and *FileWriter* writes two bytes at one time. 
+Therefore, previous program could be simply rewritten by replacing *FileOutputStream* on *FileWriter*  and *FileInputStream* on *FileReader*.
+
+![img_6.png](assets/img_6.png)
+
+There are two general-purpose byte-to-character "bridge" streams: [InputStreamReader](https://docs.oracle.com/javase/8/docs/api/java/io/InputStreamReader.html) and [OutputStreamWriter](https://docs.oracle.com/javase/8/docs/api/java/io/InputStreamReader.html).
```
Comment (egoist301) [inline stage #1/module #6. IO, NIO/lesson #1. IO/content.md:70]: You use link InputStreamReader for OutputStreamWriter. Could you fix it?

File: stage #1/module #6. IO, NIO/lesson #1. IO/content.md [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+listing the contents of a directory, and determining several common attributes of files and directories. 
+
+- It is an abstract representation of files and directory pathnames.
+- A pathname, whether abstract or in string form can be either absolute or relative. The parent of an abstract pathname may be obtained by invoking the getParent() method of this class.
+- First of all, we should create the File class object by passing the filename or directory name to it. A file system may implement restrictions to certain operations on the actual file-system object, such as reading, writing, and executing. These restrictions are collectively known as access permissions.
+- Instances of the File class are immutable; that is, once created, the abstract pathname represented by a File object will never change.
+
+Constructors of File Class:
+
+- File(File parent, String child): Creates a new File instance from a parent abstract pathname and a child pathname string.
```
Comment (egoist301) [inline stage #1/module #6. IO, NIO/lesson #1. IO/content.md:110]: Could you wrap it with ``? `File(File parent, String child)`.

File: stage #1/module #6. IO, NIO/lesson #1. IO/content.md (hunk 1 above)
Comment (Tiger656) [reply stage #1/module #6. IO, NIO/lesson #1. IO/content.md]: Done

File: stage #1/module #6. IO, NIO/lesson #1. IO/content.md (hunk 2 above)
Comment (Tiger656) [reply stage #1/module #6. IO, NIO/lesson #1. IO/content.md]: Done

File: stage #1/module #6. IO, NIO/lesson #1. IO/content.md (hunk 3 above)
Comment (Tiger656) [reply stage #1/module #6. IO, NIO/lesson #1. IO/content.md:1]: Yes, I'll do this in another PR.

File: stage #1/module #6. IO, NIO/lesson #1. IO/content.md (hunk 4 above)
Comment (Tiger656) [reply stage #1/module #6. IO, NIO/lesson #1. IO/content.md]: Fixed in another way. Table is replaced on picture with caption on.

File: stage #1/module #6. IO, NIO/lesson #1. IO/content.md (hunk 5 above)
Comment (Tiger656) [reply stage #1/module #6. IO, NIO/lesson #1. IO/content.md]: Done

File: stage #1/module #6. IO, NIO/lesson #1. IO/content.md (hunk 6 above)
Comment (Tiger656) [reply stage #1/module #6. IO, NIO/lesson #1. IO/content.md]: Fixed

File: stage #1/module #6. IO, NIO/lesson #1. IO/content.md (hunk 7 above)
Comment (Tiger656) [reply stage #1/module #6. IO, NIO/lesson #1. IO/content.md]: Wrapped

File: stage #1/module #6. IO, NIO/lesson #1. IO/content.md [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+- [Buffered Streams](https://docs.oracle.com/javase/tutorial/essential/io/buffers.html) optimize input and output by reducing the number of calls to the native API.
+- [Scanning and Formatting](https://docs.oracle.com/javase/tutorial/essential/io/scanfor.html) allows a program to read and write formatted text.
+- [I/O from the Command Line](https://docs.oracle.com/javase/tutorial/essential/io/cl.html) describes the Standard Streams and the Console object.
+- [Data Streams](https://docs.oracle.com/javase/tutorial/essential/io/datastreams.html) handle binary I/O of primitive data type and String values.
+- [Object Streams](https://docs.oracle.com/javase/tutorial/essential/io/objectstreams.html) handle binary I/O of objects.
+
+No matter how they work internally, all streams present the same simple model to programs that use them: A stream is a sequence of data. A program uses an *input stream* to read data from a source, one item at a time:
+![reading_stream_scheme.png](assets/reading_stream_scheme.png)
```
Comment (egoist301) [inline stage #1/module #6. IO, NIO/lesson #1. IO/content.md:29]: Could you please add empty line before images?

Comment (Tiger656) [reply stage #1/module #6. IO, NIO/lesson #1. IO/content.md:30]: Added

**Review by egoist301 — APPROVED** [review]
